src/test/scala/com/scalacamp/hometask/first/RetryUtilTest.scala

```diff
@@ -0,0 +1,87 @@
+package com.scalacamp.hometask.first
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.scalacamp.hometasks.first.RetryUtil.retry
+
+import scala.concurrent.duration._
+import org.scalatest.FunSuite
+import org.scalatest.Matchers._
+
+import scala.collection.mutable.ListBuffer
+
+class RetryUtilTest extends FunSuite {
+
+  case class InvocationState(invocationCount: AtomicInteger, invocationInstants: ListBuffer[Long]) {
+    def getPauseDurationsInSeconds()
+    = invocationInstants.sliding(2).map { case Seq(x, y, _*) => (y - x) / 1000 }.toList
+  }
+
+  def successOn[T]: Int => T => Boolean = {
+    var count = 0
+
+    successTime: Int =>
+      T => {
+        count = count + 1
+        if (count == successTime) true
+        else false
+      }
+  }
+
+  def decorate[T](invocationState: InvocationState, function: () => T): () => T = {
+    () => {
+      invocationState.invocationInstants += System.currentTimeMillis()
+      invocationState.invocationCount.incrementAndGet()
```

Comment:
actually you can use only invocationInstants as it's length will be invocations count

Reply:
You are right, my fault, fixed